### PR TITLE
Add an ending slash to the dir link, and don't encode the slashes.

### LINKED ISSUE #13: Lacking '/' at the end of links to directories.
I have a directory:
```
root
  |- subdir1
  |- subdir2
       |- index.html (refers to style.css with path "style.css")
       |- style.css    
```

When I start a server with `root` as its directory and visit it in the browser, the links to the subdirectories are like `<a href="http://localhost:8090/subdir2">subdir2/</a>`. (Note: there's an ending slash in the text but not in the link.)

When I go to `http://localhost:8090/subdir2`, I actually visit `subdir2/index.html`, but due to there being no ending slash in the URL, the browser thinks the ULR of the CSS file would be `http://localhost:8090/style.css`. 

To fix this relative path error, we should append a slash to its link if an item in the list is a directory.

### PR DESCRIPTION
**Pull request by KDr2**
Fix #13.

### CODE REVIEW

File: src/babashka/http_server.clj [hunk 1]
```diff
@@ -131,8 +131,18 @@
    (let [mime-types (merge default-mime-types mime-types)]
      (mime-types (filename-ext filename)))))
 
+(defn- file-link
+  "Get HTML link for a file/directory in the given dir."
+  [dir f]
+  (let [rel-path (.relativize dir f)
```
Comment (borkdude) [inline src/babashka/http_server.clj]: fs/relativize can be used here

File: src/babashka/http_server.clj [hunk 2]
```diff
@@ -131,8 +131,18 @@
...
+        ending (if (fs/directory? f) "/" "")
+        names (iterator-seq (.iterator rel-path))
```
Comment (borkdude) [inline src/babashka/http_server.clj]: Just `seq` works on a path too I think

**Comment by borkdude** [comment]
A couple of comments above. Note to self: I'll have to test if this still works on Windows before merging (of if you're able to test this yourself on Windows, go ahead).

**Comment by KDr2** [comment]
Updated.

**Comment by borkdude** [comment]
Tested on Windows and worked, thanks!

**Pull request merged**